src/components/BookDetails/BookDetails.jsx

```diff
@@ -15,10 +15,25 @@ export default function BookDetails({
 			<div className="book-details">
 				<p className="book-title">{title}</p>
 				<p className="book-author">
-					By {authors} ({publishDate})
+					{Array.isArray(authors)
+						? `By ${authors.join(', ')}`
+						: `By ${authors || 'Unknown Author'}`}{' '}
+					{publishDate
+						? `(${publishDate})`
+						: '(No publish date)'}
 				</p>
 				<p className="book-genre">
-					<span>{categories}</span>
+					{Array.isArray(categories)
```

Comment:
See comment above about making this always an array. This is a really complicated statement just to render a list of strings. We should try to reduce the complexity here!